corehq/tabs/tests/test_uitab.py

```diff
@@ -0,0 +1,82 @@
+from django.test import SimpleTestCase
+from django.template import Context, Template
+from django.core.cache.utils import make_template_fragment_key
+from django.core.cache import cache
+
+from corehq.tabs.uitab import UITab
+
+
+class TestUITab(SimpleTestCase):
+    def test_header_is_cached_correctly(self):
+        context = self._generate_context()
+
+        self._populate_cache(context)
+
+        cache_key = make_template_fragment_key(UITab.fragment_prefix_name, [context['frag_value']])
+        self.assertIsNotNone(cache.get(cache_key))
+
+    def test_header_cache_is_successfully_cleared(self):
+        context = self._generate_context()
+        self._populate_cache(context)
+
+        self._clear_dropdown_cache()
+
+        cache_key = make_template_fragment_key(UITab.fragment_prefix_name, [context['frag_value']])
+        self.assertIsNone(cache.get(cache_key))
+
+    def test_cache_accounts_for_bootstrap_5(self):
```

Comment:
I think this test might be incomplete considering the error I found in the `use_bootstrap5` boolean above. I think you should ensure that a context with `use_bootstrap=True` does not have the same cache key as `use_bootstrap5=False` otherwise we are in a world of hurt

Reply:
Interesting, I thought I had checked that, but I will make sure there is an explicit test for that

Reply:
Added a test to confirm that the bootstrap3 and 5 keys differ. That said, the reason the tests continued to pass despite the error pointed out, was because I did not add tests covering the view method itself, which is where that typo was.  I don't see a great way to add good tests for that bit of code at the moment (they'd require creating the full overhead of domain, user, etc, that obscure what was actually being tested)

Reply:
great, thank you!